lib/src/repo.rs

```diff
@@ -1345,8 +1363,13 @@ impl MutableRepo {
             rewritten_changes.insert(change_id);
         }
         for (old_commit, new_commits) in rewritten_commits {
-            for new_commit in new_commits {
-                self.record_rewritten_commit(old_commit.clone(), new_commit);
+            if new_commits.len() == 1 {
+                self.record_rewritten_commit(
+                    old_commit.clone(),
+                    new_commits.into_iter().next().unwrap(),
+                );
+            } else {
+                self.set_divergent_rewrite(old_commit.clone(), new_commits);
```

Comment:
Just rubber ducking. I wondered if "set" would work when merging more than two operation heads, but it seems okay because `rebase_descendants()` is called per `merge_operation(self, other)`.